crypto/evp/p_lib.c

```diff
@@ -855,6 +856,54 @@ EC_KEY *EVP_PKEY_get1_EC_KEY(EVP_PKEY *pkey)
         EC_KEY_up_ref(ret);
     return ret;
 }
+
+static int EVP_PKEY_set1_ECX_KEY(EVP_PKEY *pkey, int type, ECX_KEY *key)
+{
+    int ret = EVP_PKEY_assign(pkey, type, key);
+    if (ret)
+        ecx_key_up_ref(key);
+    return ret;
+}
+
+static ECX_KEY *EVP_PKEY_get0_ECX_KEY(const EVP_PKEY *pkey, int type)
+{
+    if (!evp_pkey_downgrade((EVP_PKEY *)pkey)) {
+        ERR_raise(ERR_LIB_EVP, EVP_R_INACCESSIBLE_KEY);
+        return NULL;
+    }
+    if (EVP_PKEY_base_id(pkey) != type) {
+        EVPerr(EVP_F_EVP_PKEY_GET0_ECX_KEY, EVP_R_EXPECTING_A_ECX_KEY);
```

Comment:
Why are we still propagating these flags.. And why use ERR_raise in the line above bu then EVPErr here?

Reply:
I copied code and didn't think further.  Good point

Reply:
Like I've already said, I totally expect us to make a complete sweep to change all FOOerr to ERR_raise calls at some point.  However, other stuff has higher priority for the moment.  However, you're right that *new code* shouldn't reproduce old calls if we can help it.

Reply:
It's an automated procedure, ```util/err-to-raise``` and ```util/merge-err-lines```  I argued that they should be run soon, but the project decided to wait.  Perhaps now's the time?